homeassistant/components/flux_led/__init__.py

```diff
@@ -111,14 +111,14 @@ def _async_migrator(entity_entry: er.RegistryEntry) -> dict[str, Any] | None:
         new_unique_id = None
         if entity_unique_id.startswith(entry_id):
             # Old format {entry_id}....., New format {unique_id}....
-            new_unique_id = f"{unique_id}{entity_unique_id[len(entry_id):]}"
+            new_unique_id = f"{unique_id}{entity_unique_id.removeprefix(entry_id)}"
         elif (
             ":" in entity_mac
             and entity_mac != unique_id
             and mac_matches_by_one(entity_mac, unique_id)
         ):
             # Old format {dhcp_mac}....., New format {discovery_mac}....
-            new_unique_id = f"{unique_id}{entity_unique_id[len(unique_id):]}"
+            new_unique_id = f"{unique_id}{entity_unique_id.removeprefix(unique_id)}"
```

Comment:
I think this should be
```suggestion
            new_unique_id = f"{unique_id}{entity_unique_id.removeprefix(entity_mac)}"
```

Reply:
This is a quite unclear case, maybe the intent is to remove either the unique_id or the entity_mac if they're there. Reverted in 3fd93cbbc9adf419e75398f06b01c276033b767d, as the addition of .removeprefix doesn't make the code any clearer here.